### PR TITLE
User Avatar jcrop.js Fixes

### PR DESCRIPTION
**Pull request by c5enlil**
Moved loading of assets to controller
Added space between buttons and image crop area in view for touchability
Use ONE fileset for all original files vs. one fileset per user (delete files as they pile up with no problems)
Moved creation of fileset to controller to avert mysql/i error upon first avatar upload attempt
Load proper assets for registered users who cannot view toolbar/dashboard
Other minor edits and expanded php tags

Works clean from fresh install 5.6.4.0 upon multiple tests

### CODE REVIEW

**Comment by Remo** [comment]
Why do we need to add jquery and all those core assets?

**Comment by c5enlil** [comment]
I was unable to get it working properly without loading those assets for registered users who did not have access to the dashboard/toolbar.  I was getting division by 0 errors where the widths and heights were missing from POST and the imagecreate etc was failing.  Noting that anyone with dashboard access could change avatars no problem, I gave it a try and it works.  If there's a different way to go about it I'm open to it!

**Comment by c5enlil** [comment]
Thanks for taking a look at this John.  I'll get a fix working as soon as i have a moment to do so!

**Comment by JohntheFish** [comment]
(This second coment got lost on the way, so have posted again)
Am only code reading, have not installed as the controller code is in the wrong place.

Anywhere an avatar would be edited in the core, most of the assets would already be loaded.
As far as I am aware, this should only need jcrop above what the core has already loaded. Most of the asset loading is redundant.

**Comment by c5enlil** [comment]
Working example of my legacy fork:  https://test2.juspimpin.com/index.php/register/

File: web/concrete/single_pages/profile/avatar.php [hunk 1]
```diff
@@ -64,30 +61,31 @@ function checkCoords(){
...
-                    $.each($preview, function($index, $avatar){
-                      var rx = $($avatar).parent().width() / c.w;
-                      var ry = $($avatar).parent().height() / c.h;
-                      $($avatar).css("width",Math.round(rx * boundx) + 'px')
-                        .css("height",Math.round(ry * boundy) + 'px')
-                        .css("marginLeft",-Math.round(rx * c.x) + 'px')
-                        .css("marginTop",-Math.round(ry * c.y) + 'px');
-                    });
-                  }
+                  //if (parseInt(c.w) > 0){
```
Comment (Remo) [inline web/concrete/single_pages/profile/avatar.php]: Are you uncertain about this code? Is there a chance that we might need it at some point? If not I would remove it. Having commented code without a comment about the reason for the comments will only confuse code-readers.

Comment (c5enlil) [reply web/concrete/single_pages/profile/avatar.php]: this is in removal of the placeholder and preview.   For implication purposes I left it there for the moment and could be removed to my knowledge.  The preview only worked when the user did not have an avatar set, and it didn't want to play along with AVATAR_NONE or with responsive themes (Mainly tested with Marketplace Bootstrap theme).  The two lines above are the two I can say I'm a bit unsure of.  Was getting null values and throwing mysql error on first avatar upload.  Omitting these two lines resolves the issue ?!  As long as things look good from your perspective I'll get it removes so we can merge...

Comment (c5enlil) [reply web/concrete/single_pages/profile/avatar.php]: I just started playing with 5.8 recently.  I like this implementation MUCH better than that in 5.8!

Comment (Remo) [reply web/concrete/single_pages/profile/avatar.php]: Well, I haven't debugged it and I don't know much about how it works, I just see commented code I'm not understanding ;-) I always like to understand why code is causing a problem, but it has to wait, got some pressing tasks

**Comment by c5enlil** [comment]
This tidy's the commented code and removes unneeded functions.  Still working cross theme as expected.  Code in the working example reflects these changes.

**Comment by c5enlil** [comment]
As release notes:

I've been thoroughly testing the removal of the ORIGINAL_AVATARS fileset, and the original un-cropped images within.  The files and/or the fileset itself can be removed/deleted at any time with no implications.  Fileset is re-created upon next avatar upload!

I had ambition to create a marketplace job to remove the images, but with EOL on 5.6 and it's marketplace, manual file/set removal is where it shall remain.  For the time being anyway...

**Comment by c5enlil** [comment]
It's been working well for me since I last updated.  Have at it!

**Comment by Remo** [comment]
thanks

**Pull request merged**